modules/home-environment.nix

```diff
@@ -483,6 +483,15 @@ in
         ''
     );
 
+    # Install MacOS applications to the user environment if the targetPlatform is Darwin
+    home.file."Applications/home-manager".source = let
```

Comment:
This is a bit too eager. You'll want to move the `source` attribute inside the `mkIf`. Something like
```
home.file."Applications/home-manager" =
  mkIf pkgs.stdenv.targetPlatform.isDarwin {
    source = let
      apps = pkgs.buildEnv {
        name = "home-manager-applications";
        paths = cfg.packages;
        pathsToLink = "/Applications";
      };
    in "${apps}/Applications";
  };
```

Reply:
Btw. I don't know anything about macOS but perhaps something like
```
home.file."Applications/Home Manager"
```
would be more idiomatic?

Reply:
> Btw. I don't know anything about macOS but perhaps something like
> ```
> home.file."Applications/Home Manager"
> ```
> would be more idiomatic?

I don't know, I have only been using MacOS for a week now.
The applications show up in the Launchpad which is what matters most.

I guess since _nix-darwin_ is using "Nix Apps" the folder could be renamed to "Home Manager Apps".

I assume the uninstallation process would need to remove that folder.

Reply:
> This is a bit too eager. You'll want to move the source attribute inside the mkIf

Fixed

Reply:
I renamed the folder created to "Home Manager Apps" from "home-manager"